### DISCUSSION TITLE
Coral TPU

### DISCUSSION THREAD

**Discussion by ghost** [xnorpx/blue-onyx#4] [Ideas]
Hi,
First of all, congrats and thanks so much for this tool !
I was wondering if, by any chance, you plan on supporting Coral TPUs ?

Thank you again,
Best regards!

**Comment by xnorpx** [comment]
Hi,

There is currently no support for Coral TPUS in ONNX: https://github.com/microsoft/onnxruntime/issues/10248

But even if so the models rt-detrv2 is to large to fit. 

So Blue Onyx will most likely never support Coral TPU

I would recommend to find the cheapest smallest NVidia GPU as it is much more future proof than a Coral TPU.

**Comment by Mrdindon** [comment]
Would it be the same for such TPU ? 26tops seems promising:
https://hailo.ai/products/ai-accelerators/hailo-8-ai-accelerator/#hailo8-overview

**Reply by xnorpx** [reply]
> Would it be the same for such TPU ? 26tops seems promising: https://hailo.ai/products/ai-accelerators/hailo-8-ai-accelerator/#hailo8-overview

Does not seem to have ONNX endpoint integration in ONNX. Having Blue ONNX support custom TPU is pretty unlikely unless the company implements and supports ONNX endpoint provider.

**Comment by Mrdindon** [comment]
from the feature page it seems to support ONNX
https://hailo.ai/products/ai-accelerators/hailo-8-ai-accelerator/#hailo-8-features
![image](https://github.com/user-attachments/assets/ce6e1632-5e3b-4582-9d4a-c46471cebc94)

**Reply by xnorpx** [reply]
> from the feature page it seems to support ONNX https://hailo.ai/products/ai-accelerators/hailo-8-ai-accelerator/#hailo-8-features ![image](https://private-user-images.githubusercontent.com/5002201/400863532-ce6e1632-5e3b-4582-9d4a-c46471cebc94.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzYyNzQ4NDksIm5iZiI6MTczNjI3NDU0OSwicGF0aCI6Ii81MDAyMjAxLzQwMDg2MzUzMi1jZTZlMTYzMi01ZTNiLTQ1ODItOWQ0YS1jNDY0NzFjZWJjOTQucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDEwNyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTAxMDdUMTgyOTA5WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9N2FhODk2ZDk5YTc1NjVhODM1MjYwZDdkM2I5NmRkMWEwMDFlYzIxODZkYmRlYTE5YzY0M2I0OWJiYWIwYmM1ZCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.ziL312sYq9WJIPwHbF1VQ-5ZjQQHIol3rt66V_kJfAc)

They support onnx models, which means that you give them the model and it converts to their binary format. But they don't have it integrated into onnx inference from what I can see.

https://onnxruntime.ai/docs/execution-providers/

**Comment by cgserrant** [comment]
Is the rt-detrv2 model too large for a duel coral tpu segmenting? 

https://coral.ai/docs/edgetpu/pipeline/#overview
